core/services/ardupilot_manager/mavlink_proxy/MAVProxy.py

```diff
@@ -21,17 +21,22 @@ def _get_version(self) -> Optional[str]:
         return None
 
     def assemble_command(self, master: Endpoint) -> str:
-        endpoints = " ".join([f"--out={str(endpoint)}" for endpoint in self.endpoints()])
+        endpoints_string = ""
+        for endpoint in self.endpoints():
+            if endpoint.connection_type != EndpointType.Serial:
+                endpoints_string += f" --out={str(endpoint)}"
+            else:
+                endpoints_string += f" --out={endpoint.place},{endpoint.argument}"
 
         master_string = str(master)
         if master.connection_type == EndpointType.Serial:
             if not master.argument:
                 master_string = f"{master.place}"
             else:
-                master_string = f"{master.place} --baudrate={master.argument}"
+                master_string = f"{master.place},{master.argument}"
```

Comment:
you can create a lambda to abstract the endpoint -> string conversion between serial and others, avoiding the same logic in different places.